cmd/bbolt/command_surgery_cobra.go

```diff
@@ -78,6 +81,51 @@ func surgeryRevertMetaPageFunc(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func newSurgeryCopyPageCommand() *cobra.Command {
+	copyPageCmd := &cobra.Command{
+		Use:   "copy-page <bbolt-file> [options]",
+		Short: "Copy page from the source page Id to the destination page Id",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("db file path not provided")
+			}
+			if len(args) > 1 {
+				return errors.New("too many arguments")
+			}
+			return nil
+		},
+		RunE: surgeryCopyPageFunc,
+	}
+
+	copyPageCmd.Flags().StringVar(&surgeryTargetDBFilePath, "output", "", "path to the target db file")
+	copyPageCmd.Flags().Uint64VarP(&surgerySourcePageId, "from-page", "", 0, "source page Id")
+	copyPageCmd.Flags().Uint64VarP(&surgeryDestinationPageId, "to-page", "", 0, "destination page Id")
+
+	return copyPageCmd
+}
+
+func surgeryCopyPageFunc(cmd *cobra.Command, args []string) error {
+	srcDBPath := args[0]
+
+	if surgerySourcePageId == surgeryDestinationPageId {
+		return fmt.Errorf("'--from-page' and '--to-page' have the same value: %d", surgerySourcePageId)
+	}
+
+	if err := common.CopyFile(srcDBPath, surgeryTargetDBFilePath); err != nil {
+		return fmt.Errorf("[copy-page] copy file failed: %w", err)
+	}
+
+	if err := surgeon.CopyPage(surgeryTargetDBFilePath, common.Pgid(surgerySourcePageId), common.Pgid(surgeryDestinationPageId)); err != nil {
+		return fmt.Errorf("copy-page command failed: %w", err)
+	}
```

Comment:
The struct `surgeryOptions` makes sense, but not the `(*surgeryOptions) Validate()`, because different surgery commands have different validation logic.

Let me apply the struct `surgeryOptions` together @ptabor 's comment above in a separate PR. Regarding to K8s flag pattern, I do not see any essential difference or obvious benefits. But please feel free to raise a separate issue or draft PR to discuss it. thx

Reply:
After second look, your suggestion indeed is better, it removes all the global variables. Resolved this comment in https://github.com/etcd-io/bbolt/pull/483 . thx